pkg/certs/cert-inspection/certgraphanalysis/deduplication.go

```diff
@@ -1,34 +1,68 @@
 package certgraphanalysis
 
-import "github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+import (
+	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+)
 
 func deduplicateCertKeyPairs(in []*certgraphapi.CertKeyPair) []*certgraphapi.CertKeyPair {
 	ret := []*certgraphapi.CertKeyPair{}
 	for _, currIn := range in {
-		found := false
+		if currIn == nil {
+			panic("currIn is nil")
+		}
+
+		foundIdx := -1
 		for j, currOut := range ret {
-			if currIn == nil {
-				panic("one")
-			}
 			if currOut == nil {
-				panic("two")
+				panic("currOut is nil")
 			}
-			if currOut.Name == currIn.Name {
-				ret[j] = combineSecretLocations(ret[j], currIn.Spec.SecretLocations)
-				ret[j] = combineCertOnDiskLocations(ret[j], currIn.Spec.OnDiskLocations)
-				found = true
-				break
+			// currOut has no name set - skip and allow it to be found later
+			if len(currOut.Name) == 0 {
```

Comment:
so if we hit this code path where the name of currout is empty, we should still be able to match on the `PubkeyModulus` and then merge a "better" name in, right?

Reply:
Yeah, correct. This would happen when key is found before cert, so the name cannot be extracted. Instead of backfilling the details its easier to skip the key so that it would be found on next iteration.

I'll add a comment clarifying this